kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -598,69 +598,33 @@ private void genMapCeilAxioms(Production prod, Collection<Rule> rules) {
         Seq<NonTerminal> nonterminals = elementProd.nonterminals();
         Sort sortParam = Sort(AddSortInjections.SORTPARAM_NAME, Sort("Q"));
 
-        List<K> args = new ArrayList<>();
-        for (int i = 0; i< nonterminals.length(); i++) {
-            Sort sort = nonterminals.apply(i).sort();
-            args.add(KVariable("K" + i, Att.empty().add(Sort.class, sort)));
-        } // variable arguments for MapItem (K1 is the key)
-        Seq<K> argsSeq = JavaConverters.iterableAsScalaIterable(args).toSeq();
-        K restMap = KVariable("Rest", Att.empty().add(Sort.class, mapSort));
-
-        // rule K1:KItem in_keys(MapItem(K1, K2, .., Kn) Rest:Map) => true
-        Rule inKeysRule1 = Rule(
-                KRewrite(
-                        KApply(prod.klabel().get(),
-                                args.get(0),
-                                KApply(concatProd.klabel().get(),
-                                        KApply(elementProd.klabel().get(),
-                                                argsSeq,
-                                                Att.empty()
-                                        ),
-                                        restMap
-                                )
-                        ),
-                        BooleanUtils.TRUE
-                ),
-                BooleanUtils.TRUE,
-                BooleanUtils.TRUE
-        );
-        rules.add(inKeysRule1);
-
-        // rule K1:KItem in_keys(Rest:Map) => false [owise]
-        Rule inKeysRule2 = Rule(
-                KRewrite(
-                        KApply(prod.klabel().get(),
-                                args.get(0),
-                                restMap
-                        ),
-                        BooleanUtils.FALSE
-                ),
-                BooleanUtils.TRUE,
-                BooleanUtils.TRUE,
-                Att.empty().add("owise")
-        );
-        rules.add(inKeysRule2);
+        // rule
+        //   #Ceil(MapItem(K1, K2, ..., Kn) Rest:Map)
+        // =>
+        //  {(@K1 in_keys(@Rest)) #Equals false} #And ... #And #Ceil(@Kn)
+        // Note: The {_ in_Keys(_) #Equals false} condition implies
+        // #Ceil(@K1) and #Ceil(@Rest).
```

Comment:
I agree with you that the `in_keys` condition would require the variables to be defined anyway.
Why then keep the definedness conditions?

Reply:
Only the definedness conditions `#Ceil(@K2) #And ... #And #Ceil(@Kn)` are present; I will update the comment to clarify.